### ISSUE TITLE
AzureFunctionsTools has been updated to 3.17.0, and IHttpClientFactory is no longer accepted by the constructor.

### ISSUE THREAD

**Issue by takuyateraoka** [Azure/azure-functions-core-tools#2384]
We were receiving an IHttpClientFactory in the constructor to use Named clients.
When AzureFunctionsTools was updated to 3.17.0, 
we started to receive the following message: Microsoft.Extensions.DependencyInjection.Abstractions: Unable to resolve service for type 'System.Net.Http.IHttpClientFactory'.
This error only occurs run locally.

```
public class Function1
{
    private readonly HttpClient _client;

    public Function1(IHttpClientFactory httpClientFactory)
    {
        this._client = httpClientFactory.CreateClient("named");
    }

    [FunctionName("Function1")]
    public async Task<IActionResult> Run(
        [HttpTrigger(AuthorizationLevel.Function, "get", "post", Route = null)] HttpRequest req,
        ILogger log)
    {
        log.LogInformation("C# HTTP trigger function processed a request.");
        var response = await _client.GetAsync("https://microsoft.com");
        var message = await response.Content.ReadAsStringAsync();
        return new OkObjectResult(message);
    }
}

public class Startup : FunctionsStartup
{
    public override void Configure(IFunctionsHostBuilder builder)
    {
         builder.Services.AddHttpClient("named", c => {});
    }
}
```
```
Functions:

        Function1: [GET,POST] http://localhost:7071/api/Function1

For detailed output, run func with --verbose flag.
[2021-01-13T07:10:19.815Z] Host lock lease acquired by instance ID '0000000000000000000000007E1157AD'.
[2021-01-13T07:10:21.811Z] Executed 'Function1' (Failed, Id=c95a0839-d2dd-4216-9b6c-b2b83a125027, Duration=138ms)
[2021-01-13T07:10:21.815Z] Microsoft.Extensions.DependencyInjection.Abstractions: Unable to resolve service for type 'System.Net.Http.IHttpClientFactory' while attempting to activate 'FunctionApp1.Function1'.
[2021-01-13T07:10:21.829Z] An unhandled host error has occurred.
[2021-01-13T07:10:21.831Z] Microsoft.Extensions.DependencyInjection.Abstractions: Unable to resolve service for type 'System.Net.Http.IHttpClientFactory' while attempting to activate 'FunctionApp1.Function1'.
```

**Comment by apawast** [comment]
@kashimiz could you please take a look at this?

**Comment by takuyateraoka** [comment]
This problem has been fixed in 3.24.2 (or earlier).
Thank you.
